guardian/recipes-backend
# Takedown handling

## What does this change?

Implements https://trello.com/c/QcQNtVnk

----
From the card:

**Library function to take a recipe down**

Input:

same data shape as the output from Library function to extract recipe data from an Element definition

Output:

None

You’ll need to delete the entry from the Dynamo table with a DeleteItem query (this needs both the canonical article ID and the recipe uid).  You’ll also need to use the version ID to remove the actual content from the bucket.  Finally you’ll need to call the https://trello.com/c/u9CjO8o2

S3 removal is implemented here: https://github.com/guardian/recipes-backend/tree/ag/s3-writer (https://trello.com/c/rLxRBgfq )
----

## How to test

There are automated tests for all the component parts.  There is also a manual takedown tool which can be used to test the logic "for real":

- go to the `recipes-backend-indexstore-CODE` table in DynamoDB
- choose one of the fake `capiArticleId`s in the table, preferably one with multiple recipes on it
- in a terminal, run the following:
```bash
export INDEX_TABLE=recipes-backend-indexstore-CODE
ARTICLE_ID={id-you-found} npm run dbtest
```

once the program finishes, you'll need to CTRL-C it to exit.

All of the 'recipes' from the table should be removed.

## How can we measure success?

Another step forwards
